libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountManager.java

```diff
@@ -413,7 +415,7 @@ public void signoutUser(UserAccount userAccount, Activity frontActivity, boolean
 	 * @param account Account object.
 	 * @return UserAccount object.
 	 */
-	public UserAccount buildUserAccount(Account account) {
+	public @Nullable UserAccount buildUserAccount(Account account) {
```

Comment:
The nullable annotation should have been on this method all along since it has two explicit paths to null.

In the case of this crash, the `Account` is not null or the calling method would have simply returned false with its existing guard.  So, the second null return in this method is:

```
if (authToken == null || instanceServer == null || userId == null || orgId == null) {
  return null;
}
```

That's been in place since about 2014, according to the annotated diff.  There's not much more detail as to why this was an expected condition, though.  At the least, now the calling code recognizes there is a nullable return and won't crash.

The follow-up work item in GUS would be a suitable time to investigate that logic in the wild, hopefully.  If anyone has a better suggestion to audit this logic do let me know.